src/zjs_callbacks.c

```diff
@@ -95,16 +96,23 @@ static int zjs_ringbuf_error_max = 0;
 static int zjs_ringbuf_last_error = 0;
 
 #ifdef DEBUG_BUILD
-static void set_info_string(char **str, const char *file, const char *func)
+static void set_info_string(char *str, const char *file, const char *func)
 {
-    uint32_t file_len = strlen(file);
+    char *i = (char*)(file + strlen(file));
+    while (i != file) {
+        if (*i == '/') {
+            i++;
+            break;
+        }
+        i--;
+    }
+    uint32_t file_len = strlen(i);
     uint32_t func_len = strlen(func);
-    *str = zjs_malloc(file_len + func_len + 4);
-    if (!(*str)) {
-        ERR_PRINT("failed to create file/func info string\n");
-        return;
+    uint32_t total = file_len + func_len + 4;
+    if (total > MAX_CALLER_CREATOR_LEN) {
+        total = MAX_CALLER_CREATOR_LEN;
     }
-    snprintf(*str, file_len + func_len + 4, "%s:%s()", file, func);
+    snprintf(str, file_len + func_len + 4, "%s:%s()", i, func);
```

Comment:
You meant to use `total` here; however you can just pass MAX_CALLER_CREATOR_LEN all the time instead, and not bother calculating `total`.